exercises/practice/leap/.approaches/conditional-expression/content.md

```diff
@@ -0,0 +1,114 @@
+# Conditional expression
+
+```haskell
+isLeapYear :: Integer -> Bool
+isLeapYear year =
+  if divisibleBy 100
+    then divisibleBy 400
+    else divisibleBy 4
+  where
+    divisibleBy d = year `mod` d == 0
+```
+
+
+## Conditional expressions
+
+A _conditional expression_ (`if … then … else …`) is a compound expression that uses a test to determine which of two alternatives to evaluate to.
+Many other languages feature a similar construct, often termed 'ternary operator'.
+They are also known as _`if` expressions_.
+
+Concretely, when `p` is some expression of type `Bool` and `t` and `f` are any two expressions of the same type, then `if p then t else f` will
+
+- evaluate to `t` if `p` evaluates to `True`, and
+- evaluate to `f` if `p` evaluates to `False`.
+
+Conditional expressions are [syntactic sugar][wikipedia-syntactic-sugar] for certain `case` expressions:
+
+```haskell
+_ = if p then t else f
+-- is an abbreviation of
+_ = case p of
+  True  -> t
+  False -> f
+```
```

Comment:
This might be perfect content for an `exercism/note` block.
See https://exercism.org/docs/building/markdown/markdown#h-special-blocks-sometimes-called-admonitions

Feel free to ignore